src/derive.rs

```diff
@@ -0,0 +1,239 @@
+use anyhow::anyhow;
+use std::str::FromStr;
+
+use crate::Container;
+
+/// Target object for which the trait must be derived.
+#[derive(Debug, Clone, PartialEq)]
+pub enum DeriveTarget {
```

Comment:
can technically be private now
```suggestion
/// Target object for which the trait must be derived.
#[derive(Debug, Clone, PartialEq)]
enum Target {
```